### ISSUE TITLE
Sync O365 Sharepoint reports incorrect drive_id

### ISSUE THREAD

**Issue by bwern** [abraunegg/onedrive#248]
**Attempting to sync with sharepoint driveid fails, with "ERROR: Check your 'drive_id' entry in your configuration file as it may be incorrect"**

**Application and Operating System Details:**
- OS: Linux xyz 4.18.0-11-generic #12-Ubuntu SMP Tue Oct 23 19:22:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Are you using a headless system (no gui) or with a gui installed? gui
- Application version: onedrive v1.1.1-200-g859b551
- OneDrive Account Type: Business
- DMD or LDC compiler version DMD64 D Compiler v2.083.0

**To Reproduce**
1. Followed README.Office365.md to get drive_id and configure config file
2. Run onedrive --synchronize

Application Log Output:
```
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/xyz/OneDrive
Initializing the Synchronization Engine ...

ERROR: OneDrive returned a 'HTTP 400 Bad Request' - Cannot Initialize Sync Engine
ERROR: Check your 'drive_id' entry in your configuration file as it may be incorrect

```

**Additional context**
I have confirmed that the drive_Id appears correct -- the output of the GET https://graph.microsoft.com/v1.0/drives/$drive-id/root command does report back a properly formatted id (similar to the example), with a webURL that shows the correct path to the "Shared Documents" folder I am trying to sync.

A sync attempt without the drive_id configured does operate correctly.

**Comment by abraunegg** [comment]
@bwern 
`v1.1.1-200-g859b551`
It appears you have upgraded / migrated from the 'skilion' branch, or you are using PR https://github.com/skilion/onedrive/pull/314

Have you followed the upgrade instructions to delete the items database cache?

`OneDrive Account Type: Business`

The Office365 / SharePoint support that was added, was only validated & checked against those types of accounts. Those accounts report as the following:

`Account Type: documentLibrary`

Can you please provide a few more details around the 'sync' scenario you are trying to achieve 

Can you please email the output of the following:

```
onedrive --synchronize --verbose --debug-https
```

**Comment by bwern** [comment]
Yes, migrated over from skilion some time back, updated to latest this morning. I did delete the sqllite at the time of the migration, and did it again this morning as part of these tests.

I'm not sure where to check for account type of documentLibrary, but I can confirm that the initial graph call from the example does return the following as part of it's output:

"driveType":"documentLibrary"

This sync is against a O365 business account, a team file share, being accessed/viewed through sharepoint. 

I have attached the output of the debug command, with the bearer token removed.
[bwern-onedrive.txt](https://github.com/abraunegg/onedrive/files/2613458/bwern-onedrive.txt)

**Comment by abraunegg** [comment]
@bwern 
I have gone through the log provided and nothing obvious stands out.

Can you please provide all the commands & responses as per the directions found here: https://github.com/abraunegg/onedrive/blob/master/README.Office365.md

**Comment by abraunegg** [comment]
@bwern
Any update on providing the requested information?

**Comment by bwern** [comment]
Yes, I have the output and am sanitizing the contents. Alternatively I could directly email the raw output?

**Comment by abraunegg** [comment]
You can email to: support@mynas.com.au

**Comment by ndit-dev** [comment]
i get the same problem

**Comment by abraunegg** [comment]
@bwern, @ndit-dev
Can you please run the following:

```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/260/head:pr260
git checkout pr260
make
```

When running this version, please use the following command line:

```
onedrive --synchronize --verbose
```

This should output something similar to the following:

```
Loading config ...
Using Config Dir: /home/alex/.config/onedrive
Initializing the OneDrive API ...
driveUrl: https://graph.microsoft.com/v1.0/drives/asdfasdf
itemByIdUrl: https://graph.microsoft.com/v1.0/drives/asdfasdf/items
itemByPathUrl: https://graph.microsoft.com/v1.0/drives/asdfasdf/root:/
Opening the item database ...
All operations will be performed in: /home/alex/OneDrive
Initializing the Synchronization Engine ...
Checking access token
Done checking access token
Performing a get of driveURL https://graph.microsoft.com/v1.0/drives/asdfasdf

ERROR: OneDrive returned a 'HTTP 400 Bad Request' - Cannot Initialize Sync Engine
Exception: HTTP request returned status code 400 (Bad Request)
{
    "error": {
        "code": "invalidRequest",
        "innerError": {
            "date": "2018-12-01T21:47:11",
            "request-id": "86b255f2-6320-4e2c-b1c2-55c635bbfd7a"
        },
        "message": "ObjectHandle is Invalid"
    }
}
ERROR: Check your 'drive_id' entry in your configuration file as it may be incorrect
```

Please can you provide the entire output.

**Comment by abraunegg** [comment]
@bwern, @ndit-dev
I have been able to reproduce this - will be opening an API case

**Comment by bwern** [comment]
Didn't see the update that you had reproduced so I did run the tests myself. It failed initially, but succeeded when I used the alternative drive ID that we had identified in our email conversation, the one that looked like b!v9Ba......MApt. This appears to be syncing fine in the couple of tests I have run so far.

**Comment by abraunegg** [comment]
@bwern
Can you rebuild with the specific branch as per above and run that with the working drive id ?

I am really keen on getting the right process working to get the right drive id documented

**Comment by bwern** [comment]
I ran the tests with pr260 build; failed with the original drive_id, but worked with the b!... one. I have forwarded the output from the successful run to you in email rather than try to scrub it.

**Comment by abraunegg** [comment]
@bwern, @ndit-dev
I am implementing a new switch to query for the correct drive_id for the SharePoint Library:

**Valid Name**
```
./onedrive --get-O365-drive-id ab-github-odtest
Initializing the Synchronization Engine ...
Office 365 Library Name: ab-github-odtest
drive_id: b!6H_....vVxU5
```
**Invalid Name**
```
./onedrive --get-O365-drive-id ab-github-odtest2
Initializing the Synchronization Engine ...
Office 365 Library Name: ab-github-odtest2
ERROR: This site could not be found. Please check it's name and your permissions to access the site.
./onedrive --get-O365-drive-id ab-github-odte*
Initializing the Synchronization Engine ...
Office 365 Library Name: ab-github-odte*
ERROR: This site could not be found. Please check it's name and your permissions to access the site.
```

To get this 'patch', please can you do the following and test locally. 
```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/260/head:pr260
git checkout pr260
make
```

To make this run 'correctly', your existing entry in `~/.config/onedrive/config` will need to be removed so that the queries can run against your actual account rather than your currently configured 'drive_id' entry.

What 'should' happen is - the result returned should 'match' the 'drive_id' your currently using.

**Comment by abraunegg** [comment]
Resolved by merging PR #260. Documentation updated to detail how to obtain the correct 'drive_id' using the CLI switch to configure the config file option.
